src/components/ColorPicker/CustomColorPicker.tsx

```diff
@@ -0,0 +1,138 @@
+import { ColorFormat, ColorPickerProps } from "./ColorPicker";
+
+const ColorPointer: FC = () => <div className="tw-w-4 tw-h-4 tw-rounded-full tw-border-2 tw-border-white" />;
+
+export const CustomColorPicker: FC<Omit<ColorPickerProps, "palette">> = ({ currentColor, onSelect }) => {
+    const colorFormats = Object.values(ColorFormat).map((id) => ({ id, title: id.toLocaleUpperCase() }));
+    const [colorFormat, setColorFormat] = useState(ColorFormat.Hex);
+    const [{ hsl, hsv, rgb, hex }, setColor] = useState(transformColor(currentColor));
+    useEffect(() => {
+        setColor(transformColor(currentColor));
+    }, [currentColor]);
+
+    return (
+        <div className="tw-flex tw-flex-col tw-gap-5">
+            <div className="tw-flex tw-gap-2 tw-w-full">
+                <div className="tw-relative tw-flex-grow tw-overflow-hidden tw-rounded">
+                    <Saturation
+                        hsl={hsl}
+                        hsv={hsv}
+                        pointer={() => <ColorPointer />}
+                        onChange={debounce((color) => onSelect(toColor(currentColor, { hex: toState(color).hex })))}
+                    />
+                </div>
+                <div className="tw-relative tw-w-6 tw-overflow-hidden tw-rounded">
+                    <HuePicker
+                        pointer={() => (
+                            <div className="tw-w-6 tw-flex tw-justify-center">
+                                <ColorPointer />
+                            </div>
+                        )}
+                        color={rgb}
+                        direction="vertical"
+                        width="24px"
+                        height="200px"
+                        onChange={debounce(({ hex }) => onSelect(toColor(currentColor, { hex })))}
+                    />
+                </div>
+                <div className="tw-relative tw-w-6 tw-overflow-hidden tw-rounded">
+                    <Alpha
+                        rgb={rgb}
+                        hsl={hsl}
+                        hsv={hsv}
+                        direction="vertical"
+                        width="24px"
+                        height="200px"
+                        pointer={() => (
+                            <div className="tw-w-[18px] tw-flex tw-justify-center">
+                                <ColorPointer />
+                            </div>
+                        )}
+                        style={{ pointer: { top: `${rgb.a * 100}%` } }}
+                        onChange={debounce(({ a }) => onSelect(toColor(currentColor, { rgba: { a } })))}
+                    />
+                </div>
+            </div>
+            <div className="tw-flex tw-flex-col md:tw-grid md:tw-grid-cols-3 md:tw-flex-none tw-gap-2 tw-max-w-full">
+                <div>
+                    <Dropdown
+                        menuBlocks={[{ id: "1", menuItems: colorFormats }]}
```

Comment:
I'd set a more specific id like `color-picker-format-dropdown-block` in order to prevent multiple elements with the same id, wdyt?